lib/reek/smell_detectors/base_detector.rb

```diff
@@ -122,6 +124,25 @@ def valid_detector?(detector)
           descendants.map { |descendant| descendant.to_s.split('::').last }.
             include?(detector)
         end
+
+        #
+        # @param detector_name [String] the detector in question, e.g. 'DuplicateMethodCall'
+        # @return [SmellDetector] - this will return the class, not an instance
+        #
+        def to_detector(detector_name)
+          SmellDetectors.const_get detector_name
+        end
+
+        #
+        # @return [Set<Symbol>] - all configuration keys that are available for this detector
+        #
+        def configuration_keys
+          BASIC_OPTIONS + custom_options
```

Comment:
Ah, you were referring to `default_config`, not `default_options` :)
Sorry, was a little confused by this. We also should rename this after this PR.
Good idea, updated the PR.